projects/media-viewer/src/lib/annotations/comment-set/comment/text-highlight.directive.ts

```diff
@@ -0,0 +1,46 @@
+import { AfterViewChecked, Directive, ElementRef, Input } from '@angular/core';
+import uuid from 'uuid/v4';
+
+@Directive({
+  selector: '[mvTextHighlight]'
+})
+export class TextHighlightDirective implements AfterViewChecked {
+
+  @Input() textToHighlight: string;
+  private readonly uuid = uuid();
+
+  constructor(private element: ElementRef) {}
+
+  ngAfterViewChecked(): void {
+    if (this.textToHighlight) {
+      this.highlightSearchString(this.textToHighlight);
+    }
+  }
+
+  highlightSearchString(searchString: string) {
+    this.resetHighlight();
+    this.textToHighlight = searchString;
+    const searchPattern = new RegExp(searchString, 'g');
+    const searchElement = this.element.nativeElement;
+    if (searchElement.innerHTML.includes(searchString)) {
+      searchElement.innerHTML = searchElement.innerHTML
+        .replace(searchPattern, this.highlightPattern('$&'));
+    }
+    this.textToHighlight = undefined;
+  }
+
+  resetHighlight() {
+    const searchElement = this.element.nativeElement;
+    const searchPattern = new RegExp(this.highlightPattern('(.*?)'), 'g');
+    while(searchElement.innerHTML.match(searchPattern)) {
+      const matchGroups = searchPattern.exec(searchElement.innerHTML);
+      if (matchGroups) {
+        searchElement.innerHTML = searchElement.innerHTML.replace(matchGroups[0], matchGroups[1]);
+      }
+    }
+  }
+
+  highlightPattern(dynamicText: string) {
+    return '<span id="'+this.uuid +'" style="background-color: yellow">' + dynamicText + '</span>'
```

Comment:
i think is better to add the colour in the css file rather then inline

Reply:
done